### ISSUE TITLE
Add ability to export CF output json

### ISSUE THREAD

**Issue by kenhorn** [architect/architect#1097]
It would be useful to access deployed resource details after a deployment has run - such as resource IDs / ARNs / CF generated names.

**Describe the solution you'd like**
A command line flag to export the CF outputs data as specific in sam.json (and subject to macros)

**Describe alternatives you've considered**
Every deploy could generate sam-outputs.json rather than make it optional. Another option is to chain macros *after* deploy, to allow the same mechanism to continue processing / allow arc deploy to live within a larger processing chain.

**Comment by filmaj** [comment]
Every `arc deploy` generates a `sam.json` / `sam.yaml` and there are already many (but not all) resource outputs present in the `Outputs` section of the `sam.json`. Are you looking for something more? The resolved resource IDs/ARNs/CF-generated names could be queried post-deploy via the AWS SDK CloudFormation API.

**Comment by kenhorn** [comment]
how do I access the "Ref: MyBucket"  value after a deploy from Outputs? Am I missing an existing output location?

**Comment by kenhorn** [comment]
Sorry missed the ending.. yes I could query all the resources from CF or from each API , mapping to the logical ID, to get physical ID and then parse the JSON properties. These would be great to receive in a post deploy macro though.

**Comment by filmaj** [comment]
Is `MyBucket` something you added on your own via a plugin/macro, or a reference to the `@static` bucket arc supports out of the box? The "Outputs" tab in CloudFormation for any Stacks generated by arc include three things (so far): the API Gateway ID of the app, the URL of the API Gateway deployment, and the http URL of the static bucket.

You could chain an `arc deploy` call with, say, an `aws` CLI command to get any information you want from the stack, including the out-of-the-box outputs arc provides (using the [DescribeStack](https://docs.aws.amazon.com/AWSCloudFormation/latest/APIReference/API_DescribeStacks.html) operation), e.g. 

```
$ arc deploy && AWS_REGION="us-west-2" aws cloudformation describe-stacks --stack-name MyAppNameStaging
```

**Comment by kenhorn** [comment]
Fair enough - that seems like an ok option if I wrap arc in another script
and put all the vars of interest in the Outputs. Having the whole lot in a
macro feels like a nice feature though (eg auto add all resources and
attributes to a param of an output macro). As a minimal option, outputting
the Outputs json by default feels like a natural complement to sam.json

**Comment by filmaj** [comment]
Regarding this question:

> how do I access the "Ref: MyBucket" value after a deploy from Outputs? Am I missing an existing output location?

Here's how you could extract just the static bucket URL w/ `jq` from the command-line:

```
aws cloudformation describe-stacks --stack-name MyAppStaging | jq '.Stacks[0].Outputs[] | select(.OutputKey | contains("Bucket")) | .OutputValue'
"http://myappstaging-staticbucket-hzqhgarbd5age15lhm6.s3-website.us-east-2.amazonaws.com"
```

I don't think we want `arc deploy` outputting JSON by default. It's meant as a developer tool providing a shim around working with AWS directly.

I'm going to close this issue for now.

**Comment by brianleroux** [comment]
oh! I asked for this issue sorry! I've long intended to have a `--json` flag for our CLI so you can use it programmatically not just as a user facing thing

**Comment by kenhorn** [comment]
> already many (but not all) resource outputs present in the Outputs section
of the sam.json. Are you looking for something more?

The sam.json outputs are never realised in arc land though - thanks for the
CF cli command, just thought it would be good to realise sam.outputs.json -
with possible output macro hook to continue the pipeline.
